### PR TITLE
Inline SSO javascript

### PR DESCRIPTION
**Pull request by louischan-oursky**

### CODE REVIEW

File: pkg/auth/dependency/sso/auth_handler_html.go [hunk 1]
```diff
@@ -21,28 +19,98 @@ func (i *AuthHandlerHTMLProvider) HTML() (out string, err error) {
...
+
+function cookieJarDelete(name) {
+	document.cookie = name + "=;expires=Thu, 01 Jan 1970 00:00:01 GMT;";
+}
+
+function StringStartsWith(s, search) {
+	return s.substring(0, search.length) === search;
+}
+
+function validateCallbackURL(callbackURL, authorizedURLs) {
```
Comment (kiootic) [inline pkg/auth/dependency/sso/auth_handler_html.go]: This logic does not seems matches what server does?

File: pkg/auth/dependency/sso/auth_handler_html.go [hunk 2]
```diff
@@ -21,28 +19,98 @@ func (i *AuthHandlerHTMLProvider) HTML() (out string, err error) {
...
+	if (error) {
+		windowObject.postMessage({
+			type: "error",
+			error: error
+		}, "*");
+	} else {
+		windowObject.postMessage({
+			type: "result",
+			result: result
+		}, callbackURL);
```
Comment (kiootic) [inline pkg/auth/dependency/sso/auth_handler_html.go]: I think postMessage takes allowed origin, not a URL?

File: pkg/auth/dependency/sso/auth_handler_html.go (hunk 1 above)
Comment (louischan-oursky) [reply pkg/auth/dependency/sso/auth_handler_html.go]: This is the original code.

File: pkg/auth/dependency/sso/auth_handler_html.go (hunk 2 above)
Comment (louischan-oursky) [reply pkg/auth/dependency/sso/auth_handler_html.go]: Also original code.

File: pkg/auth/dependency/sso/auth_handler_html.go (hunk 1 above)
Comment (kiootic) [reply pkg/auth/dependency/sso/auth_handler_html.go]: Do we fix it to match server logic now or later?

Comment (louischan-oursky) [reply pkg/auth/dependency/sso/auth_handler_html.go]: I am going to make the two implementation the same.

File: pkg/auth/dependency/sso/auth_handler_html.go (hunk 2 above)
Comment (louischan-oursky) [reply pkg/auth/dependency/sso/auth_handler_html.go]: https://gist.github.com/louischan-oursky/bcab4ce06c30a47f775194505f7f1aa9

Tested on Chrome.

**Comment by louischan-oursky** [comment]
@kiootic 

`validateCallbackURL` was made the same across Golang and JavaScript.

`window.postMessage` accepts URL too.

**Pull request merged**